src/main/java/net/minecraftforge/common/ForgeConfigSpec.java

```diff
@@ -626,39 +617,32 @@ public ForgeConfigSpec build()
             return ret;
         }
 
-        private void checkComment(List<String> path)
-        {
-            if (hasInvalidComment)
-            {
-                hasInvalidComment = false;
-                if (!FMLEnvironment.production)
-                {
-                    LogManager.getLogger().error(CORE, "Null comment for config option {}, this is invalid and may be disallowed in the future.",
-                            DOT_JOINER.join(path));
-                }
-            }
-        }
-
         public interface BuilderConsumer {
             void accept(Builder builder);
         }
     }
 
     private static class BuilderContext
     {
-        private @NotNull String[] comment = new String[0];
+        private final List<String> comment = new LinkedList<>();
         private String langKey;
         private Range<?> range;
         private boolean worldRestart = false;
         private Class<?> clazz;
 
-        public void setComment(String... value)
+        public void addComment(String value)
         {
-            validate(value == null, "Passed in null value for comment");
-            this.comment = value;
+            // Don't use `validate` because it throws IllegalStateException, not NullPointerException
+            Preconditions.checkNotNull(value, "Passed in null value for comment");
+
+            if (value.trim().equals("") && comment.size() == 0)
```

Comment:
You can use [`String#isBlank()`](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/String.html#isBlank()) (or at least [`String#isEmpty()`](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/String.html#isEmpty())) here.
```suggestion
            if (value.isBlank() && comment.size() == 0)
```